src/models/onnxruntime_api.h

```diff
@@ -218,12 +218,14 @@ inline void InitApi() {
       }
     }
 
-    // Search for pip installation
-    for (const std::string& lib_name : target_libraries) {
-      std::string pip_path{current_module_dir + "/../onnxruntime/capi/" + lib_name};
-      ort_lib_handle = LoadDynamicLibraryIfExists(pip_path);
-      if (ort_lib_handle != nullptr) {
-        break;
+    if (ort_lib_handle == nullptr) {
+      // Search for pip installation
+      for (const std::string& lib_name : target_libraries) {
+        std::string pip_path{current_module_dir + "/../onnxruntime/capi/" + lib_name};
```

Comment:
../onnxruntime probably doesn't exist. 
For example, some python modules may get installed in a system location like /usr/lib, then some get installed in a user location that is in the user's home folder ($HOME).

Reply:
Yeah that's a thing. I was just copying what #662 is doing. There's certainly more than one place that onnxruntime can reside.

Reply:
There might be a better solution.

Search RTLD_GLOBAL in 

https://github.com/microsoft/onnxruntime/blob/bbbaef3fa60f1ef1920af5104459251b42390135/cmake/onnxruntime_python.cmake#L320

You may turn it always ON, and let the onnxruntime_pybind11_state.so   expose all the symbols that libonnxrutime.so has. Then, in theory in another package(ONNXRuntime-GENAI) you should be able to directly use the symbols, without doing any dlopen. Because, RTLD_GLOBAL  means globally visible. Here globally means process-wide.

Reply:
The thing is, `libonnxruntime-genai.so` is loaded *before* any ORT libs. `RTLD_GLOBAL` will only expose symbol *after* ORT libs are loaded.

A better way might be directly interacting with CPython API and asked it to find `onnxruntime` for us.

Reply:
> A better way might be directly interacting with CPython API and asked it to find onnxruntime for us.

Not sure if we have the same idea, but here is one solution. We can use on linux to find the path of ort shared so. From python before importing onnxruntime_genai pybind layer, we can do something like this:

```py
import importlib.util
import ctypes
import glob

ort_package = importlib.util.find_spec("onnxruntime")
ort_package_path = ort_package.submodule_search_locations[0]
ort_lib_path = glob.glob(os.path.join(ort_package_path, "capi", "libonnxruntime.so*"))[0]
_ = ctypes.CDLL(ort_lib_path)

import onnxruntime_genai
```

We do something similar in windows by doing `os.add_dll_directory`. Since there is no linux equivalent, best is to load the shared lib and then the system will not load it again.

Changes in this PR: https://github.com/microsoft/onnxruntime-genai/pull/731